quickjs-libc.c

```diff
@@ -22,6 +22,9 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif /* _GNU_SOURCE */
```

Comment:
I guess we could also do this in `meson.build` like we do in Gum:
```
add_project_arguments('-D_GNU_SOURCE=1', language: 'c')
```

(By adding it to our existing `add_project_arguments()`.)

In this way we'll have a smaller patch to maintain on top of upstream.